bug: bytes_used_128 is underconstrained

## Bug

The prover can pick any value for x as long as 0<=x<=256**15, so it can return an incorrect length for the number of bytes required to represent a number

PoC:
```
diff --git a/src/utils/utils.cairo b/src/utils/utils.cairo
index da451281..9384374c 100644
--- a/src/utils/utils.cairo
+++ b/src/utils/utils.cairo
@@ -685,7 +685,8 @@ namespace Helpers {
         let (output) = alloc();
 
         %{
-            memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
+            # memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
+            memory[ids.output] = res = 1
             assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
         %}
         tempvar x = [output];
```

makes the test `poetry run pytest tests/src/ -m "not NoCI" --log-cli-level=INFO -n logical -vv -k test_should_return_bytes_used_in_128_word` fail with

```
    @given(word=st.integers(min_value=0, max_value=2**128 - 1))
    @settings(max_examples=20, deadline=None)
    def test_should_return_bytes_used_in_128_word(cairo_run, word):
        bytes_length = (word.bit_length() + 7) // 8
        output = cairo_run(
            "test__bytes_used_128",
            word=word,
        )
>       assert bytes_length == output[0]
E       assert 0 == 16
E       Falsifying example: test_should_return_bytes_used_in_128_word(
E           cairo_run=_factory,
E           word=0,
E       )
E       Explanation:
E           These lines were always and only run by failing examples:
E               /home/vscode/.cache/pypoetry/virtualenvs/kakarot-9tgfeqj9-py3.10/lib/python3.10/site-packages/_pytest/assertion/util.py:265
```

Asserting that, for each step, x is greater than 256**i should remediate the issue

--- a/src/utils/utils.cairo
+++ b/src/utils/utils.cairo
@@ -680,279 +680,69 @@ namespace Helpers {
     // @param value The 128-bit value.
     // @return The number of bytes used by the value.
     func bytes_used_128{range_check_ptr}(value: felt) -> felt {
-        tempvar base = 256 ** 15;
-        let bound = base;
-        tempvar max = base - 1;
-        let (output) = alloc();
-
-        %{
-            memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
-            assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
-        %}
-        tempvar x = [output];
-        [range_check_ptr] = x;
-        assert [range_check_ptr + 1] = max - x;
-
-        let range_check_ptr = range_check_ptr + 2;
-        if (value - x != 0) {
+        let (q, r) = unsigned_div_rem(value, 256 ** 15);
+        if (q != 0) {
             return 16;
         }
-        tempvar base = base / 256;
-        let bound = base;
-        tempvar max = base - 1;
-        // 1.
-        let output = output + 1;
-        %{
-            memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
-            assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
-        %}
-        tempvar x = [output];
-        [range_check_ptr] = x;
-        assert [range_check_ptr + 1] = max - x;
-
-        let range_check_ptr = range_check_ptr + 2;
-        if (value - x != 0) {
+        let (q, r) = unsigned_div_rem(value, 256 ** 14);
+        if (q != 0) {
             return 15;
         }
-        tempvar base = base / 256;
-        let bound = base;
-        tempvar max = base - 1;
-        // 2.
-        let output = output + 1;
-        %{
-            memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
-            assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
-        %}
-        tempvar x = [output];
-        [range_check_ptr] = x;
-        assert [range_check_ptr + 1] = max - x;
-
-        let range_check_ptr = range_check_ptr + 2;
-        if (value - x != 0) {
+        let (q, r) = unsigned_div_rem(value, 256 ** 13);
+        if (q != 0) {
             return 14;
         }
-        tempvar base = base / 256;
-        let bound = base;
-        tempvar max = base - 1;
-        // 3.
-        let output = output + 1;
-        %{
-            memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
-            assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
-        %}
-        tempvar x = [output];
-        [range_check_ptr] = x;
-        assert [range_check_ptr + 1] = max - x;
-
-        let range_check_ptr = range_check_ptr + 2;
-        if (value - x != 0) {
+        let (q, r) = unsigned_div_rem(value, 256 ** 12);
+        if (q != 0) {
             return 13;
         }
-        tempvar base = base / 256;
-        let bound = base;
-        tempvar max = base - 1;
-        // 0.
-        let output = output + 1;
-        %{
-            memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
-            assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
-        %}
-        tempvar x = [output];
-        [range_check_ptr] = x;
-        assert [range_check_ptr + 1] = max - x;
-
-        let range_check_ptr = range_check_ptr + 2;
-        if (value - x != 0) {
+        let (q, r) = unsigned_div_rem(value, 256 ** 11);
+        if (q != 0) {
             return 12;
         }
-        tempvar base = base / 256;
-        let bound = base;
-        tempvar max = base - 1;
-        // 1.
-        let output = output + 1;
-        %{
-            memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
-            assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
-        %}
-        tempvar x = [output];
-        [range_check_ptr] = x;
-        assert [range_check_ptr + 1] = max - x;
-
-        let range_check_ptr = range_check_ptr + 2;
-        if (value - x != 0) {
+        let (q, r) = unsigned_div_rem(value, 256 ** 10);
+        if (q != 0) {
             return 11;
         }
-        tempvar base = base / 256;
-        let bound = base;
-        tempvar max = base - 1;
-        // 2.
-        let output = output + 1;
-        %{
-            memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
-            assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
-        %}
-        tempvar x = [output];
-        [range_check_ptr] = x;
-        assert [range_check_ptr + 1] = max - x;
-
-        let range_check_ptr = range_check_ptr + 2;
-        if (value - x != 0) {
+        let (q, r) = unsigned_div_rem(value, 256 ** 9);
+        if (q != 0) {
             return 10;
         }
-        tempvar base = base / 256;
-        let bound = base;
-        tempvar max = base - 1;
-        // 3.
-        let output = output + 1;
-        %{
-            memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
-            assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
-        %}
-        tempvar x = [output];
-        [range_check_ptr] = x;
-        assert [range_check_ptr + 1] = max - x;
-
-        let range_check_ptr = range_check_ptr + 2;
-        if (value - x != 0) {
+        let (q, r) = unsigned_div_rem(value, 256 ** 8);
+        if (q != 0) {
             return 9;
         }
-        tempvar base = base / 256;
-        let bound = base;
-        tempvar max = base - 1;
-        // 0.
-        let output = output + 1;
-        %{
-            memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
-            assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
-        %}
-        tempvar x = [output];
-        [range_check_ptr] = x;
-        assert [range_check_ptr + 1] = max - x;
-
-        let range_check_ptr = range_check_ptr + 2;
-        if (value - x != 0) {
+        let (q, r) = unsigned_div_rem(value, 256 ** 7);
+        if (q != 0) {
             return 8;
         }
-        tempvar base = base / 256;
-        let bound = base;
-        tempvar max = base - 1;
-        // 1.
-        let output = output + 1;
-        %{
-            memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
-            assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
-        %}
-        tempvar x = [output];
-        [range_check_ptr] = x;
-        assert [range_check_ptr + 1] = max - x;
-
-        let range_check_ptr = range_check_ptr + 2;
-        if (value - x != 0) {
+        let (q, r) = unsigned_div_rem(value, 256 ** 6);
+        if (q != 0) {
             return 7;
         }
-        tempvar base = base / 256;
-        let bound = base;
-        tempvar max = base - 1;
-        // 2.
-        let output = output + 1;
-        %{
-            memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
-            assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
-        %}
-        tempvar x = [output];
-        [range_check_ptr] = x;
-        assert [range_check_ptr + 1] = max - x;
-
-        let range_check_ptr = range_check_ptr + 2;
-        if (value - x != 0) {
+        let (q, r) = unsigned_div_rem(value, 256 ** 5);
+        if (q != 0) {
             return 6;
         }
-        tempvar base = base / 256;
-        let bound = base;
-        tempvar max = base - 1;
-        // 3.
-        let output = output + 1;
-        %{
-            memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
-            assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
-        %}
-        tempvar x = [output];
-        [range_check_ptr] = x;
-        assert [range_check_ptr + 1] = max - x;
-
-        let range_check_ptr = range_check_ptr + 2;
-        if (value - x != 0) {
+        let (q, r) = unsigned_div_rem(value, 256 ** 4);
+        if (q != 0) {
             return 5;
         }
-        tempvar base = base / 256;
-        let bound = base;
-        tempvar max = base - 1;
-        // 0.
-        let output = output + 1;
-        %{
-            memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
-            assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
-        %}
-        tempvar x = [output];
-        [range_check_ptr] = x;
-        assert [range_check_ptr + 1] = max - x;
-
-        let range_check_ptr = range_check_ptr + 2;
-        if (value - x != 0) {
+        let (q, r) = unsigned_div_rem(value, 256 ** 3);
+        if (q != 0) {
             return 4;
         }
-        tempvar base = base / 256;
-        let bound = base;
-        tempvar max = base - 1;
-        // 1.
-        let output = output + 1;
-        %{
-            memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
-            assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
-        %}
-        tempvar x = [output];
-        [range_check_ptr] = x;
-        assert [range_check_ptr + 1] = max - x;
-
-        let range_check_ptr = range_check_ptr + 2;
-        if (value - x != 0) {
+        let (q, r) = unsigned_div_rem(value, 256 ** 2);
+        if (q != 0) {
             return 3;
         }
-        tempvar base = base / 256;
-        let bound = base;
-        tempvar max = base - 1;
-        // 2.
-        let output = output + 1;
-        %{
-            memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
-            assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
-        %}
-        tempvar x = [output];
-        [range_check_ptr] = x;
-        assert [range_check_ptr + 1] = max - x;
-
-        let range_check_ptr = range_check_ptr + 2;
-        if (value - x != 0) {
+        let (q, r) = unsigned_div_rem(value, 256 ** 1);
+        if (q != 0) {
             return 2;
         }
-        tempvar base = base / 256;
-        let bound = base;
-        tempvar max = base - 1;
-        // 3.
-        let output = output + 1;
-        %{
-            memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
-            assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
-        %}
-        tempvar x = [output];
-        [range_check_ptr] = x;
-        assert [range_check_ptr + 1] = max - x;
-
-        let range_check_ptr = range_check_ptr + 2;
-        if (value - x != 0) {
+        if (value != 0) {
             return 1;
         }
-
         return 0;
     }
 

